toolkit/tools/internal/network/network.go

```diff
@@ -65,6 +75,22 @@ func DownloadFile(url, dst string, caCerts *x509.CertPool, tlsCerts []tls.Certif
 	return
 }
 
+func removeFileIfExists(fullPath string) (err error) {
+	exists, err := file.PathExists(fullPath)
+	if err != nil {
+		err = fmt.Errorf("failed to check if file exists:\n%w", err)
+		return
+	}
+	if exists {
+		err = os.Remove(fullPath)
+		if err != nil {
+			err = fmt.Errorf("failed to remove file:\n%w", err)
+			return
+		}
+	}
+	return
+}
+
```

Comment:
I think we might not even need to check if the file exists and rely on `os.Remove` error entirely by ignoring a missing file:
```suggestion
func removeFileIgnoreMissing(fullPath string) (err error) {
	removeErr := os.Remove(fullPath).(*os.PathError)
	if removeErr != nil && removeErr.Err != syscall.ENOENT {
		return fmt.Errorf("failed to remove file (%s):\n%w", fullPath, removeErr)
	}
	return removeErr
}
```

And what do you think about making that a function inside `file.go`?

Reply:
Moved to file.go, and as per comments in the go library `if removeErr != nil && !errors.Is(removeErr, fs.ErrNotExist) {` is the modern way to check for this error.